pkg/diagnostics/systemd/analyze_logs.go

```diff
@@ -0,0 +1,128 @@
+package systemd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+
+	"github.com/openshift/origin/pkg/diagnostics/log"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+	"github.com/openshift/origin/pkg/diagnostics/types/diagnostic"
+)
+
+// AnalyzeLogs
+type AnalyzeLogs struct {
+	SystemdUnits map[string]types.SystemdUnit
+
+	Log *log.Logger
+}
+
+func (d AnalyzeLogs) Description() string {
+	return "Check for problems in systemd service logs since each service last started"
+}
+func (d AnalyzeLogs) CanRun() (bool, error) {
+	return true, nil
+}
+func (d AnalyzeLogs) Check() (bool, []log.Message, []error, []error) {
+	infos := []log.Message{}
+	warnings := []error{}
+	errors := []error{}
```

Comment:
> Instead I think I want to return a *DiagnosticResult (new type). The role of the diagnostic would be to put messages, warnings, and errors into this result and set success.

Yes, I like that idea.  I see two potential problems, but I think they're surmountable.
 1. Inability to show partial output.  If a diagnostic test gets stuck, you can't see the errors it produced along the way.
 1. Inability to interleave.  Some output is related to other output.  For instance, if you're checking a particular kubeconfig context and find two warnings and an error, you often want to display the error near the warnings.

Perhaps a DiagnosticResult could contain something like these markers to allow sorting? https://github.com/openshift/origin/blob/master/pkg/api/graph/interfaces.go#L8  Having `originator`, `severity`, `key`, `message`, `advice` would structure the output and solve problem 2.

I'd probably add high verbosity glog statements to deal with problem 1.